app/executor.php

```diff
@@ -581,57 +581,65 @@ function (string $runtimeId, array $vars, string $data, $timeout, $activeRuntime
 App::setResource('activeRuntimes', fn() => $activeRuntimes);
 
 /** Set callbacks */
-App::error(function ($utopia, $error, $request, $response) {
-    $route = $utopia->match($request);
-    logError($error, "httpError", $route);
-
-    switch ($error->getCode()) {
-        case 400: // Error allowed publicly
-        case 401: // Error allowed publicly
-        case 402: // Error allowed publicly
-        case 403: // Error allowed publicly
-        case 404: // Error allowed publicly
-        case 406: // Error allowed publicly
-        case 409: // Error allowed publicly
-        case 412: // Error allowed publicly
-        case 425: // Error allowed publicly
-        case 429: // Error allowed publicly
-        case 501: // Error allowed publicly
-        case 503: // Error allowed publicly
-            $code = $error->getCode();
-            break;
-        default:
-            $code = 500; // All other errors get the generic 500 server error status code
-    }
+App::error()
+    ->inject('utopia')
+    ->inject('error')
+    ->inject('request')
+    ->inject('response')
+    ->action(function ($utopia, $error, $request, $response) {
```

Comment:
Can we add types here?

Reply:
you mean for action? As the params are dynamic, I don't think that's easily possible?

Reply:
+1